### PR TITLE
ROC-2137 Handle non number values

### PR DESCRIPTION
**Pull request by dharmendrak**

### CODE REVIEW

File: src/main/app/forms/validation/validators/fractions.ts [hunk 1]
```diff
@@ -24,6 +24,11 @@ export class FractionsConstraint implements ValidatorConstraintInterface {
     if (value == null) {
       return true
     }
+
+    if (typeof value !== 'number') {
+      return true
```
Comment (sabahirfan) [inline src/main/app/forms/validation/validators/fractions.ts]: As this Validator is generic to be used elsewhere, does not look right to say non-numbers are valid fractions.

Comment (timja-kainos) [reply src/main/app/forms/validation/validators/fractions.ts]: not this validator's concern.
IsNumber for that

Comment (sabahirfan) [reply src/main/app/forms/validation/validators/fractions.ts]: As we are making the Fraction validator to be used with @isNumber to work properly. Doesn't feel right. But we can go with this, is not a blocker.

File: src/main/app/forms/models/amount.ts [hunk 2]
```diff
@@ -22,32 +22,42 @@ export class Amount implements Serializable<Amount> {
...
   higherValue?: number
 
-  constructor (lowerValue?: number, higherValue?: number, cannotState?: string) {
+  constructor (lowerValue?: any, higherValue?: any, cannotState?: string) {
     this.lowerValue = lowerValue
     this.higherValue = higherValue
     this.cannotState = cannotState
   }
 
+  static translateToNumber (input?: any): any {
```
Comment (geek-soft) [inline src/main/app/forms/models/amount.ts]: Wired place to have it, especially that it is public.

**Comment by geek-soft** [comment]
Despite it has been merged, in the future maybe it would make sense to keep raw form data and use it to render field value when validation failed. I haven't done any POC around this so I do not know how feasible / nice it would be. Just possible direction for the future.

**Pull request merged**